compiler/src/liveness.ml

```diff
@@ -50,7 +50,14 @@ and live_d weak d (s_o: Sv.t) =
     let s2, c2 = live_c weak c2 s_o in
     Sv.union (vars_e e) (Sv.union s1 s2), s_o, Cif(e, c1, c2)
 
-  | Cfor _ -> assert false (* Should have been removed before *)
+  | Cfor (x, r, c) ->
+    let rec loop s_o =
+      let s_o' = if weak then Sv.add (L.unloc x) s_o else s_o in
+      let s_i, c = live_c weak c s_o' in
```

Comment:
This looks wrong. `s_o'` should be `s_o` and `x` should be *removed* from `s_i`.

Reply:
Indeed, this was certainly wrong. Is this better?

Reply:
libjade fails, so I guess it's wrong.